Recurly/BaseClient.cs

```diff
@@ -48,25 +49,29 @@ public BaseClient(string apiKey)
             RestClient.AddDefaultHeader("Content-Type", "application/json");
         }
 
-        public async Task<T> MakeRequestAsync<T>(Method method, string url, Request body = null, Dictionary<string, object> queryParams = null, RequestOptions options = null, CancellationToken cancellationToken = default(CancellationToken)) where T : new()
+        public async Task<T> MakeRequestAsync<T>(Method method, string url, Request body = null, Dictionary<string, object> queryParams = null, RequestOptions options = null, CancellationToken cancellationToken = default(CancellationToken)) where T : Resource
         {
             Debug.WriteLine($"Calling {url}");
             var request = BuildRequest(method, url, body, queryParams, options);
-            var task = RestClient.ExecuteTaskAsync<T>(request, cancellationToken);
+            var task = RestClient.ExecuteAsync<T>(request, cancellationToken);
             return await task.ContinueWith(t =>
             {
                 var resp = t.Result;
                 this.HandleResponse(resp);
+                if (resp.Data is object)
```

Comment:
Shouldn't this be more specific? like `is Resource` or something?

Reply:
Updated, though `resp.Data` can only be a `Resource`